internal/cli/certificate.go

```diff
@@ -0,0 +1,324 @@
+	fs.Int32VarP(&o.ExpirationSeconds, "expiry", "e", 604800, "specify desired certificate expiration in seconds")
+	fs.StringVarP(&o.OutputFormat, "format", "f", "agent-config", "specify desired output format: either 'agent-config' to output values in one config file, or 'components' to output cert, key, and ca values in separate files")
+}
+
+func (o *CertificateOptions) Complete(cmd *cobra.Command, args []string) error {
+	return o.GlobalOptions.Complete(cmd, args)
+}
+
+func (o *CertificateOptions) Validate(args []string) error {
+	if err := o.GlobalOptions.Validate(args); err != nil {
+		return err
+	}
+
+	if len(args) == 0 {
+		return fmt.Errorf("one subcommand must be provided to 'certificate': options are: 'request'")
+	}
+
+	if args[0] != "request" {
+		return fmt.Errorf("one subcommand must be provided to 'certificate': options are: 'request'")
+	}
+
+	return nil
+}
+
+func (o *CertificateOptions) Run(ctx context.Context, args []string) error {
+	name := createUniqueName(o.Name)
+
+	// currently this is the only option
+	switch o.Scope {
+	case "enrollment":
+		o.SignerName = "ca"
+	default:
+		o.SignerName = "ca"
+	}
+
+	_, priv, err := fccrypto.NewKeyPair()
+	if err != nil {
+		return fmt.Errorf("error creating new key pair: %w", err)
+	}
+
+	csrResourceJSON, err := createCsr(o, name, priv)
+	if err != nil {
+		return fmt.Errorf("error creating csr resource: %w", err)
+	}
+
+	c, err := client.NewFromConfigFile(o.ConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("error creating client: %w", err)
+	}
+
+	err = submitCsr(name, c, ctx, csrResourceJSON)
+	if err != nil {
+		return fmt.Errorf("error submitting csr to flightctl service: %w", err)
+	}
+
+	currentCsr, err := getCsr(o, name, c, ctx)
+	if err != nil {
+		return fmt.Errorf("error reading csr %s: %w", name, err)
+	}
+
+	for !checkCsrCertReady(currentCsr) {
+		currentCsr, err = getCsr(o, name, c, ctx)
+		if err != nil {
+			return fmt.Errorf("error reading csr %s: %w", name, err)
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	// in the case of component output, we actually only need the CA data from here - we may get this
+	// data from elsewhere if we move away from the enrollmentconfig API
+	response, err := c.EnrollmentConfigWithResponse(ctx, name)
+	if err != nil {
+		return fmt.Errorf("failed to get enrollment config for %s: %w", name, err)
+	}
+	err = validateHttpResponse(response.Body, response.StatusCode(), http.StatusOK)
+	if err != nil {
+		return fmt.Errorf("failed to get enrollment config for %s: %w", name, err)
+	}
+
+	switch o.OutputFormat {
+	case "components":
+		cadata, decodeErr := base64.StdEncoding.DecodeString(response.JSON200.EnrollmentService.Service.CertificateAuthorityData)
+		if decodeErr != nil {
+			return fmt.Errorf("error decoding CA data: %w", decodeErr)
+		}
+		err = createOutputComponents(name, currentCsr, priv, cadata)
+	default:
+		err = createOutputConfig(name, c, ctx, priv, response)
```

Comment:
What should the default be? Intuitively, the command is about getting the CA bundle, certificate, and keys first, getting them in the "right" format for an agent second?

Reply:
I suppose as the main purpose for this output would be to get an agent config, I would tend to think of that as the default. But perhaps I am not aware of some other use cases.